www/codemirror/codemirror-index.html

```diff
@@ -0,0 +1,465 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset=utf8>
+  <meta name="viewport" id="viewport" content="width=device-width, user-scalable=no, initial-scale=1.0">
+  <script src="resources/lib/codemirror/lib/codemirror.js"></script>
+  <link rel="stylesheet" type="text/css" href="resources/lib/codemirror/lib/codemirror.css">
+  <script src="resources/mode/mediawiki/mediawiki.js"></script>
+  <link rel="stylesheet" type="text/css" href="resources/mode/mediawiki/mediawiki.css">
+  <link rel="stylesheet" type="text/css" href="codemirror-common.css">
+  <link rel="stylesheet" id="codemirror-theme" type="text/css" href="">
+
+  <script>
+    const applyTheme = (themeName) => {
+      document.getElementById('codemirror-theme').setAttribute('href', `codemirror-${themeName}.css`)  
+    }
+    document.addEventListener('DOMContentLoaded', () => {
+      const params = (new URL(window.location)).searchParams
```

Comment:
@joewalsh You could still add it but don't expect any effect from it unless the server side implements it. I would try to avoid varying the server response based on user settings if we can can. But I think we could still add client side changes based on it. We plan to allow the client to set the theme and platform anyways, and this could then just made part of that.

Reply:
@berndsi yeah - I mean using query params to pass information to the client-side JS, not any variance in the server response. Thinking about it some more, I would rather have the native clients always go through the same JS interface if possible instead of having the initial config through query params and any subsequent adjustments through executing JS directly.